plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -64,27 +73,56 @@ struct SoftwareModuleUpdate {
     pub path: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct TedgeConfig {
+    pub apt: AptConfig,
+}
+
+#[derive(Debug, Deserialize)]
+struct AptConfig {
+    pub name: Option<String>,
+    pub maintainer: Option<String>,
+}
+
 fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
     let status = match operation {
-        PluginOp::List {} => {
-            let apt = Command::new("apt")
-                .args(vec!["--manual-installed", "list"])
+        PluginOp::List { name, maintainer } => {
+            let dpkg_query = Command::new("dpkg-query")
+                .args(vec![
+                    "-f",
+                    "${Package}\t${Version}\t${Maintainer}\t${Status}\n",
+                    "-W",
+                ])
                 .stdout(Stdio::piped())
                 .spawn()
-                .map_err(|err| InternalError::exec_error("apt", err))?
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?
                 .wait_with_output()
-                .map_err(|err| InternalError::exec_error("apt", err))?;
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?;
 
-            let stdout = String::from_utf8(apt.stdout.as_slice().to_vec()).unwrap_or_default();
-            for line in stdout.trim_end().split('\n') {
-                if line == "Listing..." {
-                    // Ignore the first line of the stdout
-                    continue;
+            let stdout = String::from_utf8(dpkg_query.stdout).unwrap_or_default();
+
+            let filter = match (&name, &maintainer) {
+                (None, None) => Regex::new(r"install ok installed"),
+
+                _ => Regex::new(
+                    format!(
+                        r"(^{}\t.*|^\S+\t\S+\t{}\s+.*)install ok installed",
+                        name.unwrap_or_default(),
+                        maintainer.unwrap_or_default()
```

Comment:
In current scenario, if both name and maintainer are `None` value, the second pattern matching will apply. Basically, I unwrap them to default to fit all possible scenarios in one pattern (If those values unwrapped to `\S+`, you would get whole list even though you provided only one parameter)

Reply:
but if you ask about the case when both values are empty strings, then the function will produce empty list.

Reply:
Yeah, I was talking about empty `""` values only. Initially I was thinking that empty values should behave similar to `None` values. But, seeing your last comment, I agree that empty list is probably even more appropriate. This can be used as a way to filter everything out.